File: eco-core/core-plugin/src/main/java/com/willfp/ecoenchants/command/CommandEcoEnchants.java
package com.willfp.ecoenchants.command;

import com.willfp.eco.core.command.impl.PluginCommand;
import com.willfp.ecoenchants.EcoEnchantsPlugin;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class CommandEcoEnchants extends PluginCommand {
    /**
     * Instantiate a new /ecoenchants command handler.
     *
     * @param plugin The plugin for the commands to listen for.
     */
    public CommandEcoEnchants(@NotNull final EcoEnchantsPlugin plugin) {
        super(plugin, "ecoenchants", "ecoenchants.command.ecoenchants", false);

        this.addSubcommand(new CommandDebug(plugin))
                .addSubcommand(new CommandReload(plugin))
                .addSubcommand(new CommandGiverandombook(plugin))
                .addSubcommand(new CommandRandomenchant(plugin))
                .addSubcommand(new CommandLocale(plugin))
                .addSubcommand(new CommandToggleDescriptions(plugin));
    }

    @Override
    public void onExecute(@NotNull final CommandSender sender,
                          @NotNull final List<String> args) {
        sender.sendMessage(this.getPlugin().getLangYml().getMessage("invalid-command"));
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecoenchants/commands/CommandEcoEnchants.kt
package com.willfp.ecoenchants.commands

import com.willfp.eco.core.command.impl.PluginCommand
import com.willfp.ecoenchants.plugin
import org.bukkit.command.CommandSender

object CommandEcoEnchants : PluginCommand(
    plugin,
    "ecoenchants",
    "ecoenchants.command.ecoenchants",
    false
) {
    override fun onExecute(sender: CommandSender, args: List<String>) {
        sender.sendMessage(
            plugin.langYml.getMessage("invalid-command")
        )
    }

    init {
        addSubcommand(CommandReload)
            .addSubcommand(CommandToggleDescriptions)
            .addSubcommand(CommandGiveRandomBook)
            .addSubcommand(CommandGUI)
    }
}
